cdap-ui/app/cdap/components/Experiments/DetailedView/ModelsTable/index.js

```diff
@@ -291,14 +296,49 @@ const renderModel = (model, outcomeType, experimentId, newlyTrainingModel) => {
     Component = Link;
     props.to = `/ns/${getCurrentNamespace()}/experiments/create?experimentId=${experimentId}&modelId=${model.id}`;
   }
+
+  const modelStatusComp = () => {
+    if (statusIsLoading) {
+      return <IconSVG name="icon-spinner" className="fa-spin" />;
+    }
+    if (statusIsError) {
```

Comment:
Was there any specific reason why these are not part of `ModelStatusIndicator` component? We can pass in the `error`,`loading` and `getModelStatus` as props to the component which can then render the same.

Reply:
At first thought it would be cleaner to just keep `ModelStatusIndicator` as a generic status factory instead of passing in MMDS-specific error, but looking again we only have MMDS-specifc statuses here anyway, so I have made the change.